### PR TITLE
Add Prime.probably_prime?

### PR DESCRIPTION
**Pull request by chaitanyav**
method to test if a given number is probably prime
or composite using Miller-Rabin Test

### CODE REVIEW

**Comment by Arsen7** [comment]
How about a test for the method returning `false`?

**Comment by chaitanyav** [comment]
Added tests for checking some large composites.

**Comment by davydovanton** [comment]
Nice work @chaitanyav! :+1: 

But I have some style comments:
1. I think that `if` will be petty in single-line style ([link](https://github.com/bbatsov/ruby-style-guide#if-as-a-modifier))
2. You have unnecessary `return` calls in end of method ([example link](https://github.com/ruby/ruby/pull/1051/files#diff-cb376f8d9ff85c3dbf97cd71a7463898R188), [style guide link](https://github.com/bbatsov/ruby-style-guide#no-explicit-return))

Thanks :star2:

**Comment by chaitanyav** [comment]
@davydovanton Fixed the style issues.

**Comment by Arsen7** [comment]
@chaitanyav :thumbsup:

File: lib/prime.rb [hunk 1]
```diff
@@ -145,6 +145,42 @@ def prime?(value, generator = Prime::Generator23.new)
...
+  # Returns true if n is probably prime, false if n is composite
+  # == Parameters
+  #
+  #  +n+:: an arbitary integer to be checked
+  #  +k+:: optional, A parameter that determines the accuracy
+  #                  of the test
+  #  References: https://en.wikipedia.org/wiki/Miller%E2%80%93Rabin_primality_test
+  #              http://rosettacode.org/wiki/Miller-Rabin_primality_test#Ruby
+  def probably_prime?(n, k = 30)
+    raise ArgumentError, 'n should be >= 3' if n < 3
```
Comment (egilburg) [inline lib/prime.rb]: perhaps
```ruby
return false if n < 2
return true if n < 3
```

instead of special casing argument error?

Comment (chaitanyav) [inline lib/prime.rb]: @egilburg updated the code.

Comment (egilburg) [inline lib/prime.rb]: this still leaves argumenterror window if you pass in something like 2.3. while unlikely use case, I think there is value in keeping same api as `Prime.prime?` which casts to int and thus return false on `Prime.prime(2.3)`

Comment (chaitanyav) [inline lib/prime.rb]: @egilburg  What is the point of passing a float to a prime checking function and expecting a false result.

Comment (egilburg) [inline lib/prime.rb]: It's nice to keep comparable algorithms with comparable APIs.

So anyone who has currently calling `Prime.prime?(n)` should get the same data type (and presence/lack/type of exception, if appropriate) as what they would get if they switch to calling `Prime.probably_prime?(n)`, for any type and value of `n`

Comment (chaitanyav) [inline lib/prime.rb]: Its a bug in Prime.prime?, https://bugs.ruby-lang.org/issues/11606  . And i am considering to revert my patch to throw errors on invalid inputs.

**Comment by chaitanyav** [comment]
Rehearsal ---------------------------------------------------------------
Probably prime               11.990000   0.060000  12.050000 ( 12.054223)
OpenSSL::BN.prime_fasttest?   0.020000   0.000000   0.020000 (  0.024519)
----------------------------------------------------- total: 12.070000sec

                                  user     system      total        real
Probably prime               12.090000   0.050000  12.140000 ( 12.147566)
OpenSSL::BN.prime_fasttest?   0.020000   0.000000   0.020000 (  0.021217)

closing PR.

**Pull request closed without merging**